core/src/main/java/hudson/Util.java

```diff
@@ -367,10 +363,8 @@ public static boolean isSymlink(@Nonnull File file) throws IOException {
     @SuppressWarnings("NP_BOOLEAN_RETURN_NULL")
     private static Boolean isSymlinkJava7(@Nonnull File file) throws IOException {
         try {
-            Object path = File.class.getMethod("toPath").invoke(file);
-            return (Boolean) Class.forName("java.nio.file.Files").getMethod("isSymbolicLink", Class.forName("java.nio.file.Path")).invoke(null, path);
-        } catch (NoSuchMethodException x) {
-            return null; // fine, Java 6
+            Path path = file.toPath();
+            return Files.isSymbolicLink(path);
         } catch (Exception x) {
```

Comment:
only a `SecurityException` is documented to be thrown and we tend not to catch those elsewhere?.